plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -249,7 +249,7 @@ void deleteBlobs(Collection<String> blobNames) throws IOException {
             deleteBlob(blobNames.iterator().next());
             return;
         }
-        final List<Storage.Objects.Delete> deletions = new ArrayList<>();
+        final List<Storage.Objects.Delete> deletions = new ArrayList<>(blobNames.size());
```

Comment:
This is going to oversize if `blobNames.size() > MAX_BATCHING_REQUESTS` I think.